ПЕ-21б Никонов Д.О.: Создание репозитория дисциплин.

DenNikonov opened the pull request:
Создал класс DisciplineItem для хранения данных по дисциплинам.

Создал класс DisciplineRepository - интерфейс для работы с дисциплинами в БД.
Создал класс IDisciplineRepository - интерфейс репозитория дисциплин.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создание репозитория дисциплин.

Создал класс DisciplineItem для хранения данных по дисциплинам.
Создал класс DisciplineRepository - интерфейс для работы с дисциплинами в БД.
Создал класс IDisciplineRepository - интерфейс репозитория дисциплин.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создать репозиторий дисциплин.

Исправил ошибки в оформлении кода.
Создал тесты для методов созданного репозитория.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создание репозитория дисциплин.

Исправлены ошибки в коде и заменил пробелы на табуляцию в файле IDesciplineRepository.cs.
Удален и исключен из проекта файл DesciplineRepository.cs.
Включен в проект файл DesciplineRepositoryTests.cs.

dha01 commented on IS.Model/IS.Model.Tests/IS.Model.Tests.csproj:
```diff
@@ -54,6 +54,7 @@
   </Choose>
   <ItemGroup>
     <Compile Include="Properties\AssemblyInfo.cs" />
+    <Compile Include="Repository\NewDiscipline\DisciplineRepositoryTests.cs" />
```
Почему NewDiscipline ?

dha01 commented on IS.Model/IS.Model.Tests/Repository/Discipline/DisciplineRepositoryTests.cs:
```diff
@@ -0,0 +1,162 @@
+		private TransactionScope _transactionScope;
+
+		/// <summary>
+		/// Репозиторий задач.
```
Задач?

dha01 commented on IS.Model/IS.Model.Tests/Repository/Discipline/DisciplineRepositoryTests.cs:
```diff
@@ -0,0 +1,162 @@
+			_transactionScope = new TransactionScope();
+			_disciplineRepository = new DisciplineRepository();
+
+            _discipline = new DisciplineItem()
```
Отступы гуляют много где начиная отсюда.

dha01 commented on IS.Model/IS.Model.Tests/Repository/Discipline/DisciplineRepositoryTests.cs:
```diff
@@ -0,0 +1,162 @@
+			{
+				Id = 1,
+				FullName = "",
+				ShortName = "",
```
Ввести что нибудь но отличное от _disciplineNew

dha01 commented on IS.Model/IS.Model.Tests/Repository/Discipline/DisciplineRepositoryTests.cs:
```diff
@@ -0,0 +1,162 @@
+        /// </summary>
+        /// <param name="first_discipline"></param>
+        /// <param name="second_discipline"></param>
+        private void AreEqualTasks(DisciplineItem first_discipline, DisciplineItem second_discipline)
```
Tasks?

dha01 commented on IS.Model/IS.Model.Tests/Repository/Discipline/DisciplineRepositoryTests.cs:
```diff
@@ -0,0 +1,162 @@
+		/// Изменяет параметры дисциплины.
+		/// </summary>
+		[Test]
+		public void Update_Void_ReturnChangedTask()
```
Task?

dha01 commented on IS.Model/IS.Model.Tests/Repository/Discipline/DisciplineRepositoryTests.cs:
```diff
@@ -0,0 +1,162 @@
+		/// Получает список всех дисциплин.
+		/// </summary>
+		[Test]
+		public void GetList_Void_ReturnNotEmptyListWithTask()
```
Task?

dha01 commented on IS.Model/IS.Model/IS.Model.sln:
```diff
@@ -0,0 +1,22 @@
+﻿
```
Тут изменений не должно быть.

dha01 commented on IS.Model/IS.Model/Repository/Discipline/DisciplineRepository.cs:
```diff
@@ -0,0 +1,109 @@
+namespace IS.Model.Repository.Discipline
+{
+	/// <summary>
+	/// Интерфейс репозитория дисциплин.
```
Моя ошибка, это уже не интерфейс а просто репозиторий.

dha01 commented on IS.Mvc/IS.Mvc/IS.Mvc.csproj:
```diff
@@ -20,6 +20,12 @@
     <IISExpressAnonymousAuthentication />
```
Тут изменений не должно быть.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Репозиторий дисциплин.

Повторная отправка выполненного задания. Все то, что было указано в замечаниях, уже было исправлено еще в прошлый раз.
Папка NewDiscipline/ переименована в Discipline/.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Создание репозиотрия дисциплин.

Удален файл IS.Model.sln и директория NewDiscipline/.
Исправлена табуляция в DisciplineRepositoryTests.cs и ошибка в нём.

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Репозиторий дисциплин.

Откат IS.Mvc.csproj.
Исправление ошибок.

DenNikonov committed:
Merge branch 'master' into feature/Task_59_Nikonov

# Conflicts:
#	IS.Model/IS.Model.Tests/IS.Model.Tests.csproj
#	IS.Model/IS.Model/IS.Model.csproj

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Репозиторий дисциплин.

Исправлены ошибки

DenNikonov committed:
ПЕ-21б Никонов Д.О.: Репозиторий дисциплин.

Удалены ненужные строки в файле IS.Mvc.csproj
